core/src/main/java/org/infinispan/container/DefaultDataContainer.java

```diff
@@ -102,7 +108,10 @@ protected DefaultDataContainer(int concurrencyLevel, int maxEntries, EvictionStr
          default:
             throw new IllegalArgumentException("No such eviction strategy " + strategy);
       }
-      entries = new BoundedConcurrentHashMap<Object, InternalCacheEntry>(maxEntries, concurrencyLevel, eviction, evictionListener);
+
+      entries = new BoundedConcurrentHashMapV8<Object, InternalCacheEntry>(
```

Comment:
Yeah exactly, it used to :)  The V8 version has a table that resizes automatically based on number of entries.